tlv/record_type.go

```diff
@@ -141,10 +141,19 @@ func (o *OptionalRecordT[T, V]) UnwrapOrErrV(err error) (V, error) {
 }
 
 // Zero returns a zero value of the record type.
-func (t *OptionalRecordT[T, V]) Zero() RecordT[T, V] {
+func (o *OptionalRecordT[T, V]) Zero() RecordT[T, V] {
 	return ZeroRecordT[T, V]()
 }
 
+// ValOpt returns an Option of the underlying value. This can be used to chain
+// other option related methods to avoid needing to first go through the outer
+// record.
+func (o *OptionalRecordT[T, V]) ValOpt() fn.Option[V] {
+	return fn.MapOption(func(record RecordT[T, V]) V {
+		return record.Val
```

Comment:
This makes me wonder if we should extract out `Val` as a function. We have the following issue here:

```haskell
-- we have these
type OptionalRecord t v = Option (Record t v)
valOpt :: Option (Record t v) -> Option v
mapOption :: (a -> b) -> Option a -> Option b

-- if we had this
val :: Record t v -> v

-- then we could do this
valOpt :: OptionalRecord t v -> Option v
valOpt = mapOption val
```

Porting this go to is tricky since there's no (to my knowledge) easy way for us to refer to a *method* without yet supplying its receiver argument. So defining something like this wouldn't fix the ergonomics:

```go
func (r RecordT[T, V]) Val() V { ... }

func (o *OptionalRecordT[T, V]) ValOpt() fn.Option[V] {
        return fn.MapOption(RecordT.Val)(*o) // no sir
}

// but we could do this
func Val[T, V any](r Record[T, V]) V { ... }

// and then do this
func (o *OptionalRecordT[T, V] ValOpt() fn.Option[V] {
        return fn.MapOption(Val)(*o) // can do
}
```

I say all this because if we have to create new names for things every time we want to map a function over some functor, then `Map<FUNCTOR>` loses a lot of its luster.

If you have ideas and/or opinions here, lmk. This code is fine, it just points to the fact that there's room for improvement on our functional patterns.